source/fe/mapping.cc

```diff
@@ -269,6 +269,16 @@ Mapping<dim, spacedim>::InternalDataBase::InternalDataBase()
 
 
 
+template <int dim, int spacedim>
+void
+Mapping<dim, spacedim>::InternalDataBase::reinit(const UpdateFlags,
+                                                 const Quadrature<dim> &)
+{
+  DEAL_II_NOT_IMPLEMENTED();
+}
```

Comment:
Why not make the function in the base class abstract (i.e., `=0`)?

Reply:
I did think about it. Users deriving from it might then need to implement a new function, even if they do not use it. Do you think we should do it nonetheless (then I need to add an Incompatibility item in our changelog).

Reply:
But don't they have to anyway, if calling this function triggers an exception?

Reply:
Not if they implement a way for their `Mapping::get_data()` without calling the new `reinit` function and just use it in `FEValues`. This does seem like a possible scenario to me. They will get the error if they use `NonMatching::MappingInfo` with their mapping. Of course, what you said elsewhere regarding the number of users who write their own mapping class and will run into this deleted operator is likely tiny, but it could be larger than 0, and I do not want to make life for those "expert" users harder.

Reply:
Ah, I see yes. I that case, though, would `DEAL_II_ASSERT_UNREACHABLE` be the better exception?

Reply:
Yes, it would. So changed.